### PR TITLE
add RangeBitmap.eq and RangeBitmap.neq

### PR DESCRIPTION
**Pull request by richardstartin**
Adds `RangeBitmap.eq`, `RangeBitmap.eqCardinality`, `RangeBitmap.neq` and `RangeBitmap.neqCardinality`. These methods allow `RangeBitmap` to be used as a kind of compact inverted index which trades space for time.

### CODE REVIEW

File: RoaringBitmap/src/test/java/org/roaringbitmap/RangeBitmapTest.java [hunk 1]
```diff
@@ -766,6 +789,135 @@ private static void regresssionTestIssue586(long low, long high, long value) {
...
+      assertEquals(entry.getValue(), bitmap.neq(entry.getKey()));
+    }
+  }
+
+  @ParameterizedTest
+  @MethodSource("distributions")
+  public void testEqualsNotEqualsRandomWithContext(LongSupplier dist) {
+    RangeBitmap.Appender appender = RangeBitmap.appender(Long.MAX_VALUE);
+    Map<Long, RoaringBitmap> valuesToTest = new HashMap<>();
+    long max = 100_000;
```
Comment (lemire) [inline RoaringBitmap/src/test/java/org/roaringbitmap/RangeBitmapTest.java:905]: Might be an int

**Comment by lemire** [comment]
It would be cool to have a minimal example in our README (nice to have).

**Comment by richardstartin** [comment]
> It would be cool to have a minimal example in our README (nice to have).

done

**Comment by richardstartin** [comment]
I'm the only author of these files so I'm just going to merge

**Pull request merged**